Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -120,7 +118,14 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
 
         //error url's are OK as they are used to populate history on session restore.
         listData = items.filter {
-          return !($0.url.isInternalScheme && ($0.url.originalURLFromErrorPage?.isInternalScheme ?? true)) || $0.url.isAboutHomeURL
+            guard let internalUrl = InternalURL($0.url) else { return true }
+            if internalUrl.isAboutHomeURL {
+                return true
+            }
+            if let url = internalUrl.originalURLFromErrorPage, InternalURL(url) != nil {
+                return false
+            }
+            return true
```

Comment:
The original conditional was too difficult to grok, I had to make this more verbose for my sanity.